packages/quantic/force-app/main/default/lwc/quanticDidYouMean/quanticDidYouMean.js

```diff
@@ -0,0 +1,70 @@
+import { api, LightningElement, track } from 'lwc';
+import { initializeWithHeadless, registerComponentForInit } from 'c/quanticHeadlessLoader';
+
+export default class QuanticDidYouMean extends LightningElement {
+    @track state = {};
+    /** @type {()=> void} */
+    unsubscribe;
+    /**
+     * @type {Boolean}
+     */
+    @track hasQueryCorrection;
+
+    /**
+     * @type {Boolean}
+     */
+    @track wasAutomaticallyCorrected;
+
+    /**
+     * @type {string}
+     */
+    @track originalQuery;
+
+    /**
+     * @type {string}
+     */
+    @track correctedQuery;
+
+    /** @type {import("coveo").DidYouMean} */
+    didYouMean;
+    
+    /**
+     * @type {string}
+     */
+    @api engineId;
+
+    connectedCallback() {
+        registerComponentForInit(this, this.engineId);
+    }
+    
+    renderedCallback() {
+        initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+    }
+
+    /**
+     * @param {import("coveo").SearchEngine} engine
+     */
+    @api
+    initialize(engine) {
+        this.didYouMean = CoveoHeadless.buildDidYouMean(engine);
+        this.unsubscribe = this.didYouMean.subscribe(() => this.updateState());
+        
+    }
+
+    disconnectedCallback() {
+        if (this.unsubscribe) {
+          this.unsubscribe();
+        }
+      }
+
+    updateState() {
+        this.hasQueryCorrection = this.didYouMean.state.hasQueryCorrection;
+        this.wasAutomaticallyCorrected = this.didYouMean.state.wasAutomaticallyCorrected;
+        this.originalQuery = this.didYouMean.state.originalQuery;
+        this.correctedQuery = this.didYouMean.state.queryCorrection.correctedQuery;
```

Comment:
I'm curious, what else is in the `queryCorrection` that we are not using?

Reply:
`/**
 * Describe correction for a query
 */
export interface QueryCorrection {
  /**
   * The query once corrected
   */
  correctedQuery: string;
  /**
   * Array of correction for each word in the query
   */
  wordCorrections: WordCorrection[];
}

export interface WordCorrection {
  /**
   * Offset, from the beginning of the query
   */
  offset: number;
  /**
   * Length of the correction
   */
  length: number;
  /**
   * The original word that was corrected
   */
  originalWord: string;
  /**
   * The new corrected word
   */
  correctedWord: string;
}`

Reply:
`queryCorrection` contains:
`correctedQuery` :(string) The query once corrected
`wordCorrections`: (array of  `WordCorrection` ) Array of correction for each word in the query